### PR TITLE
Define some RBAC gRPCs

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/server/service/webservice/service.proto [hunk 1]
```diff
@@ -186,11 +186,26 @@ service WebService {
...
+        option (model.rbac).resource = PROJECT;
+        option (model.rbac).action = CREATE;
+    }
+    rpc UpdateProjectRBACRole(UpdateProjectRBACRoleRequest) returns (UpdateProjectRBACRoleResponse) {
+        option (model.rbac).resource = PROJECT;
+        option (model.rbac).action = UPDATE;
+    }
+    rpc DeleteProjectRBACRole(DeleteProjectRBACRoleRequest) returns (DeleteProjectRBACRoleResponse) {
+        option (model.rbac).resource = PROJECT;
+        option (model.rbac).action = DELETE;
```
Comment (nghialv) [inline pkg/app/server/service/webservice/service.proto:199]: This is not `Deleting` a Project resource, so I think `UPDATE` would be more appropriate.

File: pkg/app/server/service/webservice/service.proto [hunk 2]
```diff
@@ -186,11 +186,26 @@ service WebService {
...
+    }
+    rpc AddProjectUserGroup(AddProjectUserGroupRequest) returns (AddProjectUserGroupResponse) {
+        option (model.rbac).resource = PROJECT;
+        option (model.rbac).action = CREATE;
```
Comment (nghialv) [inline pkg/app/server/service/webservice/service.proto:203]: This too.

File: pkg/app/server/service/webservice/service.proto [hunk 3]
```diff
@@ -186,11 +186,26 @@ service WebService {
...
+    }
+    rpc DeleteProjectUserGroup(DeleteProjectUserGroupRequest) returns (DeleteProjectUserGroupResponse) {
+        option (model.rbac).resource = PROJECT;
+        option (model.rbac).action = DELETE;
```
Comment (nghialv) [inline pkg/app/server/service/webservice/service.proto:207]: And this.

Comment (knanao) [reply pkg/app/server/service/webservice/service.proto]: Right, Thank you. https://github.com/pipe-cd/pipecd/pull/3894/commits/220a06bdb28df3a7e6474cf5a3dd16c19614ccce

File: pkg/app/server/grpcapi/web_api.go [hunk 4]
```diff
@@ -1329,32 +1334,94 @@ func (a *WebAPI) GetMe(ctx context.Context, req *webservice.GetMeRequest) (*webs
...
+		return nil, err
+	}
+
+	if _, ok := a.projectsInConfig[claims.Role.ProjectId]; ok {
+		return nil, status.Error(codes.FailedPrecondition, "Failed to update a debug project specified in the control-plane configuration")
+	}
+
+	if err := a.projectStore.UpdateProjectRBACRole(ctx, claims.Role.ProjectId, req.Name, req.Policies); err != nil {
+		a.logger.Error("failed to update rbac role", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to update rbac role")
```
Comment (nghialv) [inline pkg/app/server/grpcapi/web_api.go:1368]: Should we use `gRPCEntityOperationError` to have a correct error code?

Comment (nghialv) [reply pkg/app/server/grpcapi/web_api.go:1368]: Other places as well.

Comment (knanao) [reply pkg/app/server/grpcapi/web_api.go]: True,  let me fix this.

**Review by nghialv — APPROVED** [review]
Looks nice.

File: pkg/app/server/grpcapi/web_api.go [hunk 5]
```diff
@@ -1266,7 +1271,7 @@ func (a *WebAPI) DisableStaticAdmin(ctx context.Context, req *webservice.Disable
 
 	if err := a.projectStore.DisableStaticAdmin(ctx, claims.Role.ProjectId); err != nil {
 		a.logger.Error("failed to disenable static admin login", zap.Error(err))
-		return nil, status.Error(codes.Internal, "Failed to disenable static admin login")
+		return nil, gRPCEntityOperationError(err, "disenable static admin login")
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/web_api.go:1274]:
```suggestion
		return nil, gRPCEntityOperationError(err, "disable static admin login")
```

File: pkg/app/server/grpcapi/web_api.go [hunk 6]
```diff
@@ -974,7 +979,7 @@ func (a *WebAPI) GetStageLog(ctx context.Context, req *webservice.GetStageLogReq
 	}
 	if err != nil {
 		a.logger.Error("failed to get stage logs", zap.Error(err))
-		return nil, status.Error(codes.Internal, "Failed to get stage logs")
+		return nil, gRPCEntityOperationError(err, "get stage logs")
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/web_api.go:982]: discuss on https://github.com/pipe-cd/pipecd/pull/3897#issuecomment-1237798316

File: pkg/app/server/grpcapi/web_api.go [hunk 7]
```diff
@@ -1546,7 +1613,7 @@ func (a *WebAPI) GetInsightApplicationCount(ctx context.Context, req *webservice
 			return nil, status.Error(codes.NotFound, "Not found")
 		}
 		a.logger.Error("failed to load application counts", zap.Error(err))
-		return nil, status.Error(codes.Internal, "failed to load application counts")
+		return nil, gRPCEntityOperationError(err, "load application counts")
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/web_api.go:1616]: could you remove those changes from this PR, since they're not related 🙏

Comment (knanao) [reply pkg/app/server/grpcapi/web_api.go]: Right, I changed it. 🙇

**Review by nghialv — APPROVED** [review]

**Review by khanhtc1202 — APPROVED** [review]
Here you go 🖖

**Pull request merged**